src/server/dcp_server/models.py

```diff
@@ -295,3 +297,164 @@ def eval(self, img):
+        skip_connections = []
+        for encoder in self.encoder:
+            x = encoder(x)
+            skip_connections.append(x)
+            x = self.pool(x)
+
+        x = self.bottle_neck(x)
+        skip_connections = skip_connections[::-1]
+
+        for i in np.arange(len(self.decoder), step=2):
+            x = self.decoder[i](x)
+            skip_connection = skip_connections[i//2]
+            concatenate_skip = torch.cat((skip_connection, x), dim=1)
+            x = self.decoder[i+1](concatenate_skip)
+
+        return self.output_conv(x)
+
+    def train(self, imgs, masks):
+       
+        lr = self.train_config['lr']
+        epochs = self.train_config['n_epochs']
+        batch_size = self.train_config['batch_size']
+        # optimizer_class = self.train_config['optimizer']
+
+        # Convert input images and labels to tensors
+
+        # normalize images 
+        imgs = [(img-np.min(img))/(np.max(img)-np.min(img)) for img in imgs]
+        # convert to tensor
+        imgs = torch.stack([torch.from_numpy(img.astype(np.float32)) for img in imgs])
+        imgs = imgs.unsqueeze(1) if imgs.ndim == 3 else imgs
+        # Your classification label mask
+        masks = np.array(masks)
+        masks = torch.stack([torch.from_numpy(mask[1]) for mask in masks])
+
+        # Create a training dataset and dataloader
+        train_dataset = TensorDataset(imgs, masks)
+        train_dataloader = DataLoader(train_dataset, batch_size=batch_size)
+
+        loss_fn = nn.CrossEntropyLoss()
+        optimizer = Adam(params=self.parameters(), lr=lr)
+
+        for _ in tqdm(range(epochs), desc="Running UNet training"):
+
+            self.loss = 0
+
+            for _, (imgs, masks) in enumerate(train_dataloader):
+                imgs = imgs.float()
+                masks = masks.long()
+                
+                #forward path
+                preds = self.forward(imgs)
+                loss = loss_fn(preds, masks)
+                
+                #backward path
+                optimizer.zero_grad()
+                loss.backward()
+                optimizer.step()
+            
+            self.loss /= len(train_dataloader) 
+
+    def eval(self, img):
+        """
+        Evaluate the model on the provided image and return the predicted label.
+            Input:
+            img: np.ndarray[np.uint8]
+            Output: y_hat - The predicted label
+        """ 
+        # normalise
+        img = (img-np.min(img))/(np.max(img)-np.min(img))
+        img = torch.from_numpy(img).float().unsqueeze(0)
+
+        img = img.unsqueeze(1) if img.ndim == 3 else img
+        # convert to tensor
+        # img = torch.permute(torch.tensor(img.astype(np.float32)), (2, 0, 1)).unsqueeze(0) 
+
+        preds = self.forward(img)
+        y_labels = torch.argmax(preds, 1).numpy()[0]
+
+        y_instances = label((y_labels > 0).astype(int))[0]
+
+        y_hat = np.stack((y_instances, y_labels))
+
+        return y_hat
```

Comment:
can you rename these variables (y_hat, y_instances etc.) to the variable names we used in CellPosePatchCNN, so we are consistent in our naming?